src/search/SearchBar.ts

```diff
@@ -150,7 +150,9 @@ export class SearchBar implements Component<SearchBarAttrs> {
 										!indexState.failedIndexingUpTo &&
 										currentResult &&
 										this._state().indexState.progress !== 0 &&
-										indexState.progress === 0
+										indexState.progress === 0 &&
+										//if period is changed from search view a new search is triggered there,  and we do not want to overwrite its result
+										!this.timePeriodChanged(currentResult.restriction.end, indexState.aimedMailIndexTimestamp)
```

Comment:
this only prevents triggering the search from the search bar. Do we need to take care of triggering the search via URL, e.g. going back? We can also address it separately but I wanted to bring it to your attention

Reply:
updating the url only triggers a search if `SearchModel.isNewSearch` is true, so it should be fine I think.